src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -397,6 +398,9 @@ void TemporalMemory::learnOnSegments(
           segment, presynapticCell, initialPermanence_);
       }
     }
+
+    if (_connections.dataForSegment(segment).synapses.size() == 0)
```

Comment:
Still use braces to mark the if/loop statements even when there is only a single statement.